templates/Template.NEP5.CSharp/NEP5.Crowdsale.cs

```diff
@@ -0,0 +1,70 @@
+using Neo.SmartContract.Framework;
+using Neo.SmartContract.Framework.Services.Neo;
+using Neo.SmartContract.Framework.Services.System;
+using System;
+using System.Numerics;
+
+namespace Template.NEP5.CSharp
+{
+    public partial class NEP5 : SmartContract
+    {
+        private static BigInteger GetTransactionAmount(object state)
+        {
+            var notification = (object[])state;
+            // Checks notification format
+            if (notification.Length != 4) return 0;
+            // Only allow Transfer notifications
+            if ((string)notification[0] != "Transfer") return 0;
+            // Check dest
+            if ((byte[])notification[2] != ExecutionEngine.ExecutingScriptHash) return 0;
+            // Amount
+            return (BigInteger)notification[3];
+        }
+
+        private static bool Mint()
+        {
+            if (Runtime.InvocationCounter != 1)
+                throw new Exception();
+
+            var notifications = Runtime.GetNotifications();
+            if (notifications.Length == 0)
+                throw new Exception();
+
+            BigInteger neo = 0;
+            BigInteger gas = 0;
+
+            for (int i = 0; i < notifications.Length; i++)
+            {
+                var notification = notifications[i];
+
+                if (notification.ScriptHash == NeoToken())
+                {
+                    neo += GetTransactionAmount(notification.State);
+                }
+                else if (notification.ScriptHash == GasToken())
+                {
+                    gas += GetTransactionAmount(notification.State);
+                }
+            }
+
+            StorageMap contract = Storage.CurrentContext.CreateMap(GetStoragePrefixContract());
+            var current_supply = contract.Get("totalSupply").ToBigInteger();
```

Comment:
Use AsBigInteger() will report error if contract.Get("totalSupply") is null.
```suggestion
            var totalSupplyData = contract.Get("totalSupply");
            if (totalSupplyData != null && totalSupplyData.Length > 0)
            {
                current_supply = totalSupplyData.AsBigInteger();
            }
```

Reply:
I did not change this so if the deployment had not been done, 'totalSupply' would fail.
Do you think it is better to return zero when is not yet supply?

Reply:
Anyway, I've followed your advice and added it to `Mint` and `TotalSupply` to have a uniform behavior.